src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -903,7 +927,7 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
             NodeAutoCompleteUtilities.PostAutoLayoutNodes(wsViewModel.DynamoViewModel.CurrentSpace, node.NodeModel, clusterNodesModel.Select(x => x.NodeModel), false, false, false, finalizer);
 
             //record all node and wire creation as one undo
-            DynamoModel.RecordUndoModels(wsViewModel.Model, newNodesAndWires);
+            DynamoModel.RecordUndoModels(wsViewModel.Model, newNodesAndWires.Where(n => n != null).ToList());
```

Comment:
So how can this be null ? Is DynamoModel.MakeConnectionCommand silently failing without throwing on mismatched connections ? If that's true then is there a way to identify the failure before doing anything else ?

Reply:
I am not sure how it is adding a null at this point. Let me see.